support/Environments/caltech_hpc_gcc.sh

```diff
@@ -0,0 +1,312 @@
+        cd libxsmm
+        make CXX=g++ CC=gcc FC=gfortran -j4
+        cd $dep_dir
+        rm libxsmm.tar.gz
+        echo "Installed LIBXSMM into $dep_dir/libxsmm"
+        cat >$dep_dir/modules/libxsmm <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/libxsmm/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/libxsmm/lib"
+prepend-path CPATH "$dep_dir/libxsmm/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/libxsmm/"
+EOF
+    fi
+    cd $dep_dir
+
+    if [ -f $dep_dir/yaml-cpp/lib/libyaml-cpp.a ]; then
+        echo "yaml-cpp is already installed"
+    else
+        echo "Installing yaml-cpp..."
+        wget https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.6.2.tar.gz -O yaml-cpp.tar.gz
+        tar -xzf yaml-cpp.tar.gz
+        mv yaml-cpp-* yaml-cpp-build
+        cd $dep_dir/yaml-cpp-build
+        mkdir build
+        cd build
+        cmake -D CMAKE_BUILD_TYPE=Release -D YAML_CPP_BUILD_TESTS=OFF \
+              -D CMAKE_C_COMPILER=gcc -D CMAKE_CXX_COMPILER=g++ \
+              -D YAML_CPP_BUILD_CONTRIB=OFF \
+              -D YAML_CPP_BUILD_TOOLS=ON \
+              -D CMAKE_INSTALL_PREFIX=$dep_dir/yaml-cpp ..
+        make -j4
+        make install
+        cd $dep_dir
+        rm -r yaml-cpp-build
+        rm -r yaml-cpp.tar.gz
+        echo "Installed yaml-cpp into $dep_dir/yaml-cpp"
+        cat >$dep_dir/modules/yaml-cpp <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/yaml-cpp/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/yaml-cpp/lib"
+prepend-path CPATH "$dep_dir/yaml-cpp/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/yaml-cpp/"
+EOF
+    fi
+    cd $dep_dir
+
+    if [ -f $dep_dir/libsharp/lib/libsharp.a ]; then
+        echo "libsharp is already installed"
+    else
+        echo "Installing libsharp..."
+        wget https://github.com/Libsharp/libsharp/archive/v1.0.0.tar.gz -O libsharp.tar.gz
+        tar -xzf libsharp.tar.gz
+        mv libsharp-* libsharp_build
+        cd $dep_dir/libsharp_build
+        autoconf
+        ./configure --prefix=$dep_dir/libsharp --disable-openmp
+        make -j4
+        mv ./auto $dep_dir/libsharp
+        cd $dep_dir
+        rm -r libsharp_build
+        rm libsharp.tar.gz
+        echo "Installed libsharp into $dep_dir/libsharp"
+        cat >$dep_dir/modules/libsharp <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/libsharp/lib"
+prepend-path LD_LIBRARY_PATH "$dep_dir/libsharp/lib"
+prepend-path CPATH "$dep_dir/libsharp/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/libsharp/"
+EOF
+    fi
+    cd $dep_dir
+
+    # Set up Charm++ because that can be difficult
+    if [ -f $dep_dir/charm/verbs-linux-x86_64-smp/lib/libck.a ]; then
+        echo "Charm++ is already installed"
+    else
+        echo "Installing Charm++..."
+        wget https://github.com/UIUC-PPL/charm/archive/v6.10.2.tar.gz
+        tar xzf v6.10.2.tar.gz
+        mv charm-6.10.2 charm
+        cd $dep_dir/charm
+        ./build charm++ verbs-linux-x86_64-smp --with-production -j4
```

Comment:
I think you can delete this line (it's redundant with the next line)

Reply:
I'm unsure about what is intended from charm regarding that, actually -- the help text in the build script made me concerned -- from LIBS was `compile additional parallel libraries with Charm++ core`, and the 'with' there seemed ambiguous about whether it uses the charm core to make the compile the additional libs, or if it intended 'in addition'. I guess it seems to work with just the `LIBS` build, so maybe that's fine?
I included this mostly for safety in case that interpretation wasn't entirely correct.